### PR TITLE
Bug 1467401 - convert _.groupBy() usages to ES6

### PR DESCRIPTION
**Pull request by eliottvincent**

### CODE REVIEW

File: ui/js/models/repository.js [hunk 1]
```diff
@@ -32,7 +32,11 @@ treeherder.factory('ThRepositoryModel', [
 
         const getOrderedRepoGroups = function () {
             if (!Object.keys(orderedRepoGroups).length) {
-                const groups = _.groupBy($rootScope.repos, function (r) { return r.repository_group.name; });
+                // group repos by 'repository_group.name'
```
Comment (camd) [inline ui/js/models/repository.js]: We could save a line with spread:

    const groups = $rootScope.repos.reduce((acc, repo, i, a, group = v => v.repository_group.name) => (
      { ...acc, [group(repo)]: [...acc[group(repo)] || [], repo] }
    ), {});

And slightly less terse names will make it easier to interpret, I think.  Would you mind doing them all like this?

Comment (eliottvincent) [reply ui/js/models/repository.js]: Thanks! I'll do that.

**Comment by eliottvincent** [comment]
Travis failed because of a download error. Can you please trigger it again? @camd @edmorley 
(btw, is there a way to manually trigger a new Travis build without bothering you guys?)

**Comment by edmorley** [comment]
Sadly Travis doesn't give a way to allow all users the ability to retrigger - instead they limit it to just users with commit access.

**Pull request merged**